### ISSUE TITLE
Empty browser after moving into a new maildir

### ISSUE THREAD

**Issue by Ornanovitch** [neomutt/neomutt#1947]
Hi,
I'm not sure this issue is related with [1903](https://github.com/neomutt/neomutt/issues/1903) and/or [1887](https://github.com/neomutt/neomutt/issues/1887), but it is also related to the display of folders. More precisely to the browser : 
 
## Expected Behaviour

Start Neomutt with my default profile (My muttrc starts with `source ~/.config/mutt/accounts/account1.muttrc`). Then : 

- `s ?` toggles the browser to save a message to another folder in the correct maildir
- in the editor, `a ?` toggles the browser to search for a file to attach
- source to another account show the related maildir (`source .../account2.muttrc`)
- `s ?` toggles the browser to save a message to another folder in the correct maildir
- in the editor, `a ?` toggles the browser to search for a file to attach
- etc. with account3

## Actual Behaviour

Start Neomutt with my default profile (My muttrc starts with `source ~/.config/mutt/accounts/account1.muttrc`). :heavy_check_mark: 

Then : 

- `s ?` toggles the browser to save a message to another folder in the correct maildir :heavy_check_mark: 
- in the editor, `a ?` toggles the browser to search for a file to attach :heavy_check_mark: 
- source to another account show the related maildir (`source .../account2.muttrc`) :heavy_check_mark: 
- `s ?` toggles the browser but with an empty list :negative_squared_cross_mark: (unless I use `tab` to switch to mailboxes view)
- in the editor, `a ?` toggles the browser but with an empty list :negative_squared_cross_mark: 
- etc. with account3, and if I go back to account1, I have the same problem.

Neomutt does not crashes

## Steps to Reproduce

- Make two different accounts muttrc
- In your principal muttrc, source one of your account muttrc as the default one
- Start neomutt and source the other account muttrc (or map a macro to do that)
- Try to save a message or attach a file

## How often does this happen?

- Always

## When did it start to happen?

- When I upgraded to 2019-10-25 with the Manjaro Community repo

## NeoMutt Version

```
NeoMutt 2019-10-25
Copyright (C) 1996-2016 Michael R. Elkins et autres.
NeoMutt ne fournit ABSOLUMENT AUCUNE GARANTIE ; pour les détails tapez 'neomutt -vv'.
NeoMutt est un logiciel libre, et vous êtes libre de le redistribuer
sous certaines conditions ; tapez 'neomutt -vv' pour les détails.

System: Linux 5.3.8-3-MANJARO (x86_64)
ncurses: ncurses 6.1.20180127 (compiled with 6.1.20180127)
libidn2: 2.2.0 (compiled with 2.2.0)
GPGme: 1.13.1
hcache backends: kyotocabinet, gdbm, lmdb

Compiler:
Using built-in specs.
COLLECT_GCC=cc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++ --enable-shared --enable-threads=posix --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --enable-checking=release --enable-default-pie --enable-default-ssp --enable-cet=auto
Thread model: posix
gcc version 9.2.0 (GCC) 

Configure options: --prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --gpgme --enable-lua --notmuch --gss --ssl --sasl --with-ui=ncurses --with-idn2=/usr --idn=0 --idn2=1 --lmdb --kyotocabinet --gdbm

Compilation CFLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -std=c99 -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Options de compilation :
  -autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo -gnutls +gpgme +gss +hcache -homespool +idn +inotify 
  -locales_hack +lua +meta -mixmaster +nls +notmuch +openssl +pgp +sasl +smime 
  -sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

I used mutt-wizard to config my accounts months ago, everything worked great untill this new bug
I don't use tmux nor screen
I use local folders with isync

In the neomuttdebug0 (with level 3), I noticed those two lines (but I can't tell at what moment they appear, they are lost between dozens lines in the same second) : 
```

[2019-11-07 11:28:37]<2> hcache_kyotocabinet_open() kcdbopen failed for /home/ornanovitch/.cache/mutt-wizard/gmail/headers#type=kct#opts=lc#rcomp=lex: open failed (file not found) (ecode 3)
[2019-11-07 11:28:37]<1> mx_path_probe() unable to stat !: No such file or directory (errno 2)
```

**Comment by Ornanovitch** [comment]
Just realized that there was a new release with a lot of corrections. I wait for Manjaro repo to propose the update and I'll tell you if it changes anything :)

**Comment by Ornanovitch** [comment]
Solved with 20191111 release !

Thanks a lot for your work
